src/AppBundle/Service/ReportService.php

```diff
@@ -53,6 +53,12 @@ public function setReportTypeBasedOnCasrec(Report $report)
             return;
         }
 
+        if($report instanceof Report and $report->getType() !== null)
```

Comment:
the method type hints "(Report $report)", so if a non-report is passed to the method, it'll throw a fatal exception and die. You could omit "$report instanceof Report and " if you want